src/chunker/chunk_scalar_mul.rs

```diff
@@ -49,31 +49,129 @@ pub fn chunk_hinted_scalar_mul_by_constant<T: BCAssigner>(
         let depth = min(Fr::N_BITS - i, i_step);
         // double(step-size) point
         if i > 0 {
-            let double_coeff = coeff_iter.next().unwrap();
-            let step = step_p_iter.next().unwrap();
-            let point_after_double = trace_iter.next().unwrap();
+            // divide this part into three to fit into 4mb chunk and 1000 stack limit
+            if depth > i_step / 3 {
+                for _ in 0..(i_step / 3) {
+                    let double_coeff = coeff_iter.next().unwrap();
+                    let step = step_p_iter.next().unwrap();
+                    let point_after_double = trace_iter.next().unwrap();
+    
+                    let (double_loop_script, double_hints) = G1Affine::hinted_check_double(c, double_coeff.0, double_coeff.1);
+    
+                    loop_scripts.push(double_loop_script);
+                    hints.extend(double_hints);
+    
+                    c = (c + c).into_affine();
+                }
 
-            let (double_loop_script, doulbe_hints) =
-                G1Affine::hinted_check_add(c, *step, double_coeff.0, double_coeff.1);
+                let mut segment_script = script! {};
+                for script in loop_scripts.clone() {
+                    segment_script = segment_script.push_script(script.compile());
+                }
+                loop_scripts.clear();
+
+                let mut update = G1PointType::new(assigner, &format!("{}_{}_piece_1", prefix, i));
+                update.fill_with_data(crate::chunker::elements::DataType::G1PointData(c));
+                let segment = Segment::new_with_name(format!("{}_loop_{}_piece_1", prefix, i), segment_script)
+                    .add_parameter(&type_acc)
+                    .add_result(&update)
+                    .add_hint(hints.clone());
+                hints.clear();
+                segments.push(segment);
+
+                type_acc = update;
+
+                for _ in (i_step / 3)..(2 * i_step / 3) {
```

Comment:
could there be a case where depth > (i_step/3) but less than (2* i_step / 3) ? in that case it seems like execution of this loop will not be necessary.
for i_step = 12, it will not be the case, as the final loop has depth = 2 (=254 - 12n) so control won't flow to this branch so is not a problem.

Reply:
yes you are right. I knew it will not be problem for i_step=12 as you indicated but better include all possible cases properly. done.